exporters/otlp/all/src/main/java/io/opentelemetry/exporter/otlp/metrics/OtlpGrpcMetricExporterBuilder.java

```diff
@@ -121,15 +126,22 @@ public OtlpGrpcMetricExporterBuilder setEndpoint(String endpoint) {
   }
 
   /**
-   * Sets the method used to compress payloads. If unset, compression is disabled. Currently
-   * supported compression methods include "gzip" and "none".
+   * Sets the method used to compress payloads. If unset, compression is disabled. Compression
+   * method "gzip" and "none" are supported out of the box. Support for additional compression
+   * methods is available by implementing {@link Compressor} and {@link CompressorProvider}.
    */
   public OtlpGrpcMetricExporterBuilder setCompression(String compressionMethod) {
     requireNonNull(compressionMethod, "compressionMethod");
+    if (compressionMethod.equals("none")) {
+      delegate.setCompression(null);
+      return this;
+    }
+    Set<String> supportedCompressionMethods = CompressorUtil.supportedCompressors();
     checkArgument(
-        compressionMethod.equals("gzip") || compressionMethod.equals("none"),
-        "Unsupported compression method. Supported compression methods include: gzip, none.");
-    delegate.setCompression(compressionMethod);
+        supportedCompressionMethods.contains(compressionMethod),
+        "Unsupported compressionMethod. Compression method must be \"none\" or one of: "
+            + supportedCompressionMethods.stream().collect(joining(",", "[", "]")));
```

Comment:
This bit is essentially duplicated across the 3 signal implementations. It might be rad to instead remove `supportedCompressors()` in favor of another method called `CompressorUtil.checkSupported(method)` and reusing it.

Reply:
> This bit is essentially duplicated across the 3 signal implementations.

6 implementations actually.. In the past we've debated about the merits of duplicating validation logic in an effort to make the rules for transparent to users. I.e. you can immediately see how validation is performed rather than stepping into a helper function. I think in this case though, the logic is complex and we can make it clear that validation is performed elsewhere by carefully naming the methods.